buildSrc/build.gradle.kts

```diff
@@ -82,24 +81,58 @@ extra["intellijReleaseType"] = when {
 }
 
 extra["versions.androidDxSources"] = "5.0.0_r2"
-
 extra["customDepsOrg"] = "kotlin.build"
 
 repositories {
     jcenter()
     maven("https://jetbrains.bintray.com/intellij-third-party-dependencies/")
     maven("https://maven.pkg.jetbrains.space/kotlin/p/kotlin/kotlin-dependencies")
+    maven("https://kotlin.bintray.com/kotlinx")
+    maven("https://kotlin.bintray.com/kotlin-dev")
     gradlePluginPortal()
 
     extra["bootstrapKotlinRepo"]?.let {
         maven(url = it)
     }
 }
 
+val generateCompilerVersion by tasks.registering(VersionGenerator::class) {}
```

Comment:
Version is artifact which used everywhere: in compiler, build scripts and so on, generated class is single object representing version. Do you mean teamcity builds or localbuilds?

Reply:
It prevents reuse of buildSrc compilation for all builds. Previously it was built once and used from remote cache until it is changed again.
https://ge.jetbrains.com/s/3ews3emjjzveu/timeline?details=b2ro2byz5cuke&expanded=WyIxMiJd
Also would be great to load version from resource file to avoid changing class files at all

Reply:
I'd suggest to continue this discussion in scope of unifying version process, after initial merge.

Reply:
@vvlevchenko  We should fix it before merge, and fix looks pretty simple. The problem with your local build is it can't reuse cached version built by TeamCity

Reply:
Do you mean by "fix" replace class generation with string in resources?

Reply:
Yes, you may keep the class just don't alter it every build, load version from resource file, it should fix the issue with caching